pkg/scheduler/eventhandlers_test.go

```diff
@@ -261,3 +266,84 @@ func TestNodeConditionsChanged(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdatePodInCache(t *testing.T) {
+	ttl := 10 * time.Second
+	nodeName := "node"
+
+	tests := []struct {
+		name   string
+		oldObj interface{}
+		newObj interface{}
+	}{
+		{
+			name:   "test update pod with same uid",
+			oldObj: makeBasePod(t, nodeName, "oldUID", "testName", "100m", "500", "", []v1.ContainerPort{{HostIP: "127.0.0.1", HostPort: 80, Protocol: "TCP"}}),
+			newObj: makeBasePod(t, nodeName, "oldUID", "testName", "100m", "500", "", []v1.ContainerPort{{HostIP: "127.0.0.1", HostPort: 8080, Protocol: "TCP"}}),
+		},
+		{
+			name:   "test update pod with different uid",
+			oldObj: makeBasePod(t, nodeName, "oldUID", "testName", "100m", "500", "", []v1.ContainerPort{{HostIP: "127.0.0.1", HostPort: 80, Protocol: "TCP"}}),
+			newObj: makeBasePod(t, nodeName, "newUID", "testName", "100m", "500", "", []v1.ContainerPort{{HostIP: "127.0.0.1", HostPort: 8080, Protocol: "TCP"}}),
+		},
+	}
+	for _, tt := range tests {
+		stopCh := make(chan struct{})
+		schedulerCache := cache.New(ttl, stopCh)
+		schedulerQueue := queue.NewPriorityQueue(nil)
+		defer close(stopCh)
+		t.Run(tt.name, func(t *testing.T) {
+			sched := &Scheduler{
+				SchedulerCache:  schedulerCache,
+				SchedulingQueue: schedulerQueue,
+			}
+			sched.addPodToCache(tt.oldObj)
+			sched.updatePodInCache(tt.oldObj, tt.newObj)
+			pod, err := sched.SchedulerCache.GetPod(tt.newObj.(*v1.Pod))
+			if err != nil {
+				t.Errorf("Test case %q failed: should get pod from scheduler after update with error %v, got %v", tt.name, nil, err)
+			}
+			if pod.UID != tt.newObj.(*v1.Pod).UID {
+				t.Errorf("Test case %q failed: should get pod from scheduler after update with pod UID %v, got %v", tt.name, tt.newObj.(*v1.Pod).UID, pod.UID)
+			}
+
+		})
+	}
+}
+
+type testingMode interface {
```

Comment:
This doesn't look necessary to me. You can simply pass `*testing.T` to `makeBasePod` and then `t.Fatalf` there, if necessary.